plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/data/LayoutMetaDataService.java

```diff
@@ -29,11 +40,50 @@
  * @kieler.rating yellow 2012-10-09 review KI-25 by chsch, bdu
  * @author msp
  */
-public class LayoutMetaDataService {
+public final class LayoutMetaDataService {
 
     /** the layout data service instance, which is created lazily. */
     private static LayoutMetaDataService instance;
 
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private LayoutMetaDataService() {
+        
+        // every class that implements IDataObject 
+        // or is used for internal properties must be registered here
```

Comment:
I don't think so since the way an object is created/clone cannot really be derived from the meta data.